src/build-config.ts

```diff
@@ -397,22 +447,52 @@ async function buildConfig(
     tsOptions,
     dts,
   )
-  const outputExports = getExportConditionDist(pkg, exportCondition, cwd)
+  const outputExports: Array<string | { format: 'cjs' | 'esm'; file: string }> =
+    dts
+      ? getExportTypeDist(exportCondition, cwd)
+      : getExportConditionDist(pkg, exportCondition, cwd)
+
+  if (pkg.bin) {
```

Comment:
Once we have the `src/bin` convention, will this be easier? So we can align all the exports conditions